decidim-initiatives/app/commands/decidim/initiatives/validate_sms_code.rb

```diff
@@ -29,11 +29,11 @@ def call
       private
 
       def verification_metadata_valid?
-        @verification_metadata && @verification_metadata["verification_code"].present?
+        @verification_metadata && @verification_metadata[:verification_code].present?
```

Comment:
Without this, we have a failing spec in the SMS initiative signature:

https://github.com/decidim/decidim/actions/runs/8344693593/job/22837924440

```
Failures:

  1) Initiative signing when personal data collection is enabled when the user has not signed the initiative yet an signs it when the user fills phone number with valid authorization and inserts correct phone number and inserts the correct code number the vote is created
     Failure/Error: expect(page).to have_content("initiative has been successfully signed")
       expected to find text "initiative has been successfully signed" in "Skip to main content\nMaggio, Harvey and Bauch\nHelp\nMeetings\nActivity\nYour verification code does not match ours. Please double-check the SMS we sent you.\nHome\nMain menu\n/\nInitiatives\nSign <script>alert(\"initiative_title\");</script> Dolor facere similique. 6490\nComplete your data\nMobile phone number\nSMS code verification\nFinish\nCheck the SMS received at your phone\nVerification code*\nRequired field\nCheck code and continue\nWelcome to Maggio, Harvey and Bauch participatory platform.\nalert(\"organization_description\"); Molestias perferendis in. 6466\nDecidim\nHome\nInitiatives\nHelp\nResources\nActivity\nMeetings\nDownload Open Data files\nMy account\nConfiguration\nMy public profile\nNotifications\nConversations\nTerms and Conditions\nCookie settings\nMaggio, Harvey and Bauch at X\nMaggio, Harvey and Bauch at Facebook\nMaggio, Harvey and Bauch at Instagram\nMaggio, Harvey and Bauch at YouTube\nMaggio, Harvey and Bauch at GitHub\nEnglish\nChoose language\nWebsite made with free software\n(External link)\n.\nCreative Commons License"

     [Screenshot Image]: file:///home/runner/work/decidim/decidim/spec/decidim_dummy_app/tmp/capybara/failures_r_spec_example_groups_initiative_signing_when_personal_data_collection_is_enabled_when_the_user_has_not_signed_the_initiative_yet_an_signs_it_when_the_user_fills_phone_number_with_valid_authorization__584.png

     [Screenshot HTML]: file:///home/runner/work/decidim/decidim/spec/decidim_dummy_app/tmp/capybara/failures_r_spec_example_groups_initiative_signing_when_personal_data_collection_is_enabled_when_the_user_has_not_signed_the_initiative_yet_an_signs_it_when_the_user_fills_phone_number_with_valid_authorization__584.html


     # ./spec/system/initiative_signing_sms_verification_spec.rb:141:in `block (8 levels) in <top (required)>'

```